prototypes/vof_advection/vof_advection.cc

```diff
@@ -0,0 +1,2790 @@
+                                     (eta_n - eta_nm1 + 1e-16);
+
+                secant_update = -delta_volume_n / (delta_volume_prime + 1e-16);
+                eta_np1       = eta_n + secant_update;
+
+                eta_nm1                = eta_n;
+                eta_n                  = eta_np1;
+                inside_cell_volume_nm1 = inside_cell_volume_n;
+                delta_volume_nm1       = delta_volume_n;
+              } // End secant method loop.
+
+            if (secant_it >= 20)
+              {
+                eta_n = 0.0;
+              }
+
+            std::vector<types::global_dof_index> dof_indices(dofs_per_cell);
+            cell->get_dof_indices(dof_indices);
+
+            // L2 projection of the cell-wise (discontinuous) correction to have
+            // a continuous correction at the dofs.
+            for (unsigned int i = 0; i < dofs_per_cell; ++i)
+              {
+                volume_correction(dof_indices[i]) +=
+                  eta_n * n_cells_per_dofs_inv;
+              }
+          }
+      } // End loop on cells.
+
+    volume_correction.compress(VectorOperation::add);
+    volume_correction.update_ghost_values();
+  }
+
+  template <int dim>
+  void
+  SignedDistanceSolver<dim>::conserve_global_volume(
+    const MPI_Comm &mpi_communicator)
+  {
+    /* We want to find a global correction function to apply to the dof value of
+    the signed_distance so that the geometric global volume encompased by the
+    level 0 of the signed_distance V_d and by the iso-contour 0.5 of the phase
+    fraction V_VOF match. This is required because the computed distance doesn't
+    belong to the Q1 approximation space. We solve the non-linear problem:
+    DeltaV(xi) = V_VOF - V_d(xi) = 0, where  xi = C*eta is the correction
+    function on the signed_distance that we are looking for, with eta being the
+    cell wise correction compute with compute_cell_wise_volume_correction() and
+    C being a constant. We use the secant method to do so. See Ausas et al.
+    (2012) for more details.*/
+
+    /* Compute targetted global volume. It corresponds to the one enclosed by
+    the level 0 of the level_set vector (same volume as the one enclosed
+    byiso-contour 0.5 of the phase fraction).*/
+    const double global_volume =
+      compute_volume(mapping, dof_handler, fe, level_set, mpi_communicator);
+
+    /* Initialization of values for the secant method. The subsricpt nm1 (or n
+    minus 1) stands for the previous secant iteration (it = n-1), the
+    subsricpt n stands for the current iteration and the subsricpt np1 stands
+    for the next iteration (it = n+1).*/
+    double global_volume_nm1 = 0.0;
+    double global_volume_n   = 0.0;
+
+    double global_delta_volume_nm1 = 0.0;
+    double global_delta_volume_n   = 0.0;
+
+    double global_delta_volume_prime = 0.0;
+
+    double C_nm1 = 0.0;
+    double C_n   = 0.0;
+    double C_np1 = 0.0;
```

Comment:
Can you either give a more descriptive variable name or provide a comment that says what C is?